aptos-move/framework/src/natives/account.rs

```diff
@@ -3,28 +3,38 @@
 
 use move_deps::{
     move_binary_format::errors::PartialVMResult,
-    move_core_types::{account_address::AccountAddress, gas_schedule::GasCost},
-    move_vm_runtime::native_functions::NativeContext,
+    move_core_types::account_address::AccountAddress,
+    move_vm_runtime::native_functions::{NativeContext, NativeFunction},
     move_vm_types::{
-        gas_schedule::NativeCostIndex,
-        loaded_data::runtime_types::Type,
-        natives::function::{native_gas, NativeResult},
-        pop_arg,
-        values::Value,
+        loaded_data::runtime_types::Type, natives::function::NativeResult, pop_arg, values::Value,
     },
 };
 use smallvec::smallvec;
 use std::collections::VecDeque;
+use std::sync::Arc;
 
-pub fn native_create_address(
+/***************************************************************************************************
+ * native fun create_address
+ *
+ *   gas cost: base_cost
+ *
+ **************************************************************************************************/
+#[derive(Debug, Clone)]
+pub struct CreateAddressGasParameters {
+    pub base_cost: u64,
+}
+
```

Comment:
It seems like these are pretty constant typed... is there a way to unify these into a GasParameters struct or something that distinguishes the handful of types?

Reply:
There are pros and cons around such unification. (This was also discussed on the PR in move-language.) I'm tending to some symmetric boilerplate and dupllocation here. This might eventually be easier to maintain and extend instead of trying to optimize this.

Reply:
Yeah it's a choice between having less boilerplate code or more extensibility. If we feel this is a bit too much, we can always make fine adjustments in the future.